### PR TITLE
Make bench progress callback async

### PR DESCRIPTION
**Pull request by littletof**
related #6102, not closing.

### CODE REVIEW

**Comment by littletof** [comment]
@ry forgot about that. Added a test, and updated readme too.

File: std/testing/bench_test.ts [hunk 1]
```diff
@@ -338,6 +338,30 @@ test({
...
+    const asyncCallbacks = [];
+
+    await runBenchmarks(
+      { silent: true },
+      (progress) => {
+        return new Promise(resolve => {
+          setTimeout(() => {
+            asyncCallbacks.push(progress);
+            resolve();
+          }, 10);
```
Comment (ry) [inline std/testing/bench_test.ts:356]: Maybe use window.queueMicrotask?

Comment (littletof) [reply std/testing/bench_test.ts:356]: Ok, will change it.

Comment (littletof) [reply std/testing/bench_test.ts]: One question, is simply `queueMicrotask()` okay, or should i add the `window.` as well? Havent used `queueMicrotask` before, tests check out without window.

Comment (ry) [reply std/testing/bench_test.ts]: leaving out the window is preferable - just specified that to be clear that it's global

**Comment by littletof** [comment]
@ry should be fixed now.

File: std/testing/bench_test.ts (hunk 1 above)
Comment (littletof) [reply std/testing/bench_test.ts]: Understood, thanks!

**Pull request merged**